cartographer/mapping/2d/submap_2d.cc

```diff
@@ -127,22 +127,22 @@ void Submap2D::Finish() {
 }
 
 ActiveSubmaps2D::ActiveSubmaps2D(const proto::SubmapsOptions2D& options)
-    : options_(options), range_data_inserter_(CreateRangeDataInserter()) {
-  // We always want to have at least one likelihood field which we can return,
-  // and will create it at the origin in absence of a better choice.
-  AddSubmap(Eigen::Vector2f::Zero());
-}
+    : options_(options), range_data_inserter_(CreateRangeDataInserter()) {}
 
 std::vector<std::shared_ptr<Submap2D>> ActiveSubmaps2D::submaps() const {
   return submaps_;
 }
 
 void ActiveSubmaps2D::InsertRangeData(const sensor::RangeData& range_data) {
```

Comment:
The general API is a bit strange that the outside class uses the submap() accessor to determine to which submaps these points were added - the caller needs to know implementation details of this class (which is the reason this PR requires changing the trajectory builder implementation).
How about we'd change the interface of this method (and in 3D) to actually return the affected submaps directly? Maybe that's something for another PR though.

Reply:
I see your point. I think it's up to the definition: When is a submap considered "active". At least for now, the submap() getter will return those submaps the range data was inserted last time. @wohe do you have any opinion on that?

Reply:
Seems reasonable to me. Return the submaps directly from the insertion function. We still need the separate `submaps()` function in this case since it is also used to determine which submap to use for scan matching.